pkg/gcp/validation/validation.go

```diff
@@ -112,6 +113,12 @@ func validateGCPDisks(disks []*api.GCPDisk, fldPath *field.Path) []error {
 		if disk.Boot && "" == disk.Image {
 			allErrs = append(allErrs, field.Required(idxPath.Child("image"), "image is required for boot disk"))
 		}
+		if disk.KmsKeyName != "" && disk.KmsKeyServiceAccount == "" {
```

Comment:
same as above and this time for both `KmsKeyName` and `KmsKeyServiceAccount` as well.